### DISCUSSION TITLE
KVM HA feature too slow

### DISCUSSION THREAD

**Discussion by baltazorbest** [apache/cloudstack#10097] [Q&A] [answered]
Hey!
I have the Cloudstack version 4.19.1.3 with 3 management hosts and 5 compute kvm nodes. I need to figure out how to enable HA in my infrastructure.
My primary and secondary storages are NFS. 
I enabled HA on KVM hosts and set kvm provider KVMHAProvider.
Then I created the compute offering with HA feature and a few instances with HA offering.
After that I turned off one kvm host (to simulate sudden power cable unplug) and expected that the server would change its status to ‘down’ soon but it happened roughly 30 minutes later.

I changed several global settings and restarted cloudstack-management one by one but it didn’t help.
The changes were:

- alert.warm: 1800 -> 120
- ping.interval: 60 -> 10
- kvm.ha.degraded.max.period: 300 -> 120

What should I do to force the ‘failed’ host to change its status to ‘down’ after 2 or 3 minutes and make the instances from this host to start on other hosts?
Thank you in advance.

**Comment by DaanHoogland** [comment]
@baltazorbest , there are several other settings , like `kvm.ha.activity.check.interval`, `kvm.ha.activity.check.max.attempts`, etc. search for ".ha." in the global settings tab to find more.

Keep in mind that you need to avoid the chance on a split brain scenario. take care

**Comment by rp-** [comment]
This https://github.com/apache/cloudstack/pull/9677 might also be interesting for you.

I think with your installed version, you can only lower the host down if you set the `wait` configuration lower.

**Comment by baltazorbest** [comment, marked as answer]
Hey.

Thanks.
After changing the wait parameter from 1800 seconds to 20 seconds, I see a difference, and now the host changes its status to Down after 5 minutes.
